### DISCUSSION TITLE
Edge Functions: request entity too large

### DISCUSSION THREAD

**Discussion by martinrobaey5th** [supabase/supabase#20864] [Questions]
Dear,

I encounter a problem trying to deploy a little simple function to our project.
I use the google api to create an event in an agenda. Locally it works. When I try to deploy to the server I get this error "request entity too large". 
I understand that my script seems to be too big (indeed the content-length of the function being pushed is 12651518 Bytes). Could you help to reduce the size of the script ? 
Even though I find my script so simple that I find it odd to be so big. I am only importing these two lib :
import { google } from 'npm:googleapis';
import { JWT } from 'npm:google-auth-library';

Thanks in advance.

**Comment by laktek** [comment]
@martinrobaey5th We currently do have a max limit for Edge Functions of 10MB. Your function size is around 12MB.

One of the things you can try is only import the submodules you need from `googleapis`. Their [installation documentation](https://www.npmjs.com/package/googleapis#installation) has a list of submodules, so you can try to pick and import only the ones you need.

This will also help to reduce the boot time of your function.

**Comment by ardabeyazoglu** [comment]
Is there a way to inspect what takes so much space in edge function build ? so we can try to remove big packages..
